Assets/_BForBoss/_Core/Scripts/Enemies/FloatingTargetBehaviour.cs

```diff
@@ -63,15 +73,49 @@ protected override void LifeCycleFinished()
             {
                 gameObject.SetActive(false);
             }
+            ToggleKnockbackState(false);
         }
 
         private void Awake()
         {
+            _rb = GetComponent<Rigidbody>();
+            if (_rb != null)
+            {
+                _rb.isKinematic = true;
+                _rb.useGravity = false;
+            }
             if (_healthbar == null)
             {                
                 Debug.LogWarning("A FloatingTargetBehaviour is missing a health bar");
             }
         }
+
+        private void CheckKnockback(float time)
+        {
+            if (_rb == null) 
+                return;
+            _knockbackCurrentTime += time;
+            if (_knockbackCurrentTime > _knockbackMaxDuration || _rb.velocity.sqrMagnitude < float.Epsilon * float.Epsilon)
```

Comment:
interesting, why `float.Epsilon * float.Epsilon` ?

Reply:
using squared magnitude and comparing against a squared value is slightly more performant than comparing against magnitude itself. Since epsilon is just a small value placeholder I don't know that it matters that much though

Reply:
Fair enough, could declare it as a Const/Readonly variable? to avoid doing the multiplication each time